From 520b3c43f93011423e71b343654f5851871dc92b Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:46 -0400
Subject: [scsi] hpsa: fix bad comparison of signed with unsigned in hpsa_update_scsi_devices

Message-id: <1410186734-19986-56-git-send-email-thenzl@redhat.com>
Patchwork-id: 92087
O-Subject: [RHEL6.6 e-stor PATCH 55/83] hpsa: fix bad comparison of signed with unsigned in hpsa_update_scsi_devices
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

rescan_hba_mode was defined as a u8 so could never be less than zero:

        rescan_hba_mode = hpsa_hba_mode_enabled(h);
        if (rescan_hba_mode < 0)
                goto out;

Signed-off-by: Joe Handzik <joseph.t.handzik@hp.com>
Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 2bbf5c7f9fe8fea2b2ccf02b9b5d2cbd997f2de1)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 3d147e1..ae9b6c9 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -3026,7 +3026,7 @@ static void hpsa_update_scsi_devices(struct ctlr_info *h, int hostno)
  int reportlunsize = sizeof(*physdev_list) + HPSA_MAX_PHYS_LUN * 24;
  int i, n_ext_target_devs, ndevs_to_allocate;
  int raid_ctlr_position;
- u8 rescan_hba_mode;
+ int rescan_hba_mode;
  DECLARE_BITMAP(lunzerobits, MAX_EXT_TARGETS);
 
  currentsd = kzalloc(sizeof(*currentsd) * HPSA_MAX_DEVICES, GFP_KERNEL);
-- 
1.7.1